### PR TITLE
fix(agentless): allow disabling check update batching and document ba…

### PR DESCRIPTION
**Pull request by santoshpulluri**
### Summary
This change adds an explicit way to disable agentless check update batching and documents the public configuration for batch_flush_interval.

### Problem
Agentless batching was introduced recently, but there was no public-facing README documentation for:

how to configure batch_flush_interval
what values are valid
how lower or higher values affect behavior
There was also no explicit way to disable batching once enable_agentless = true. A value of 0 fell back to the default flush interval instead of turning batching off.

### Changes
treat batch_flush_interval = 0 as an explicit opt-out for agentless batching
preserve the default 500ms behavior when the setting is omitted
document batch_flush_interval in the README configuration reference
add README guidance for:
default behavior
valid Go duration values
0s disabling buffering
tradeoffs of lower vs higher values
add test coverage for the zero-value behavior
### Behavior After This Change
omitted batch_flush_interval: defaults to 500ms
batch_flush_interval = "0s": disables batching and sends updates immediately
positive duration values: enable batching with the configured flush interval
negative values: continue to fall back to the default interval
Test### ing
added coverage for the zero-value batching behavior
ran the agentless batching test file successfully

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -307,6 +313,45 @@ critical_threshold = 0
 
 [HCL]: https://github.com/hashicorp/hcl "HashiCorp Configuration Language (HCL)"
 
+### Batch Flushing for Agentless Check Updates
```
Comment (kswap) [inline README.md:316]: Should we add all these details in the documentation rather than ReadMe ? We can point to the doc in the readme.

**Review by shashankNandigama — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (santoshpulluri) [reply README.md:316]: Yes. The documentation was never updated for Agent-less setup for ESM. 
We need a blog/documentation about the Agent-less setup & cover other details like how node distribution affects the performance of ESM cluster etc.
We will take up the documentation work separately in the month of May 2026.
